overlord/state/notices.go

```diff
@@ -211,6 +219,29 @@ type AddNoticeOptions struct {
 
 	// Time, if set, overrides time.Now() as the notice occurrence time.
 	Time time.Time
+
+	// RepeatCheck, if set, overrides "RepeatAfter" value dynamically based
+	// on the current state.
+	//
+	// NOTE: Set/Get operation on "data" are persisted and can be used
+	// for statful tracking throughout notice's lifetime.
+	RepeatCheck func(data InternalNoticeData) (newRepeatAfter time.Duration, err error)
+}
+
+type InternalNoticeData struct {
```

Comment:
this feels awkward, more than just using interface{}

Reply:
If we unmarshal to interface{} we are limited by [`json.Unmarshal`](https://pkg.go.dev/encoding/json#Unmarshal) supported types:
```
bool, for JSON booleans
float64, for JSON numbers
string, for JSON strings
[]interface{}, for JSON arrays
map[string]interface{}, for JSON objects
nil for JSON null
```
Which would be confusing on the consumer side when type casting the data interface{} fails unexpectedly because it was unmarshalled into a `map[string]interface{}` instead of the used struct.

An alternative would be to have a pre-defined repeat-check-data-type per notice-type, like:
```go
type ChangeUpdateNoticeRepeatCheckData struct {
    lastStatus Status  `json:"last-status"`
}

type WarningNoticeRepeatCheckData struct {
    ...
}
```

Another alternative is passing an unmarshal callback
```go
RepeatCheck func(unmarshal func(v interface{}) error) (newRepeatAfter time.Duration, newRepeatCheckData interface{}, err error)
```